File: frontend/src/app/shared/service/admin/common/NewsAdmin.service.ts
import { Injectable } from '@angular/core';
import {HttpClient} from '@angular/common/http';
import {BehaviorSubject, Observable} from 'rxjs';

import {environment} from 'src/environments/environment';

import {NewsDto} from 'src/app/shared/model/common/News.model';
import {NewsCriteria} from 'src/app/shared/criteria/common/NewsCriteria.model';
import {AbstractService} from 'src/app/zynerator/service/AbstractService';


@Injectable({
  providedIn: 'root'
})
export class NewsAdminService extends AbstractService<NewsDto, NewsCriteria> {
     constructor(private http: HttpClient) {
        super();
        this.setHttp(http);
    }

    public constrcutDto(): NewsDto {
        return new NewsDto();
    }

    public constrcutCriteria(): NewsCriteria {
        return new NewsCriteria();
    }

    get API() {
        return environment.apiUrlZynservice + 'admin/news/';
    }
}

File: backend/src/main/java/ma/zs/alc/service/facade/admin/common/NewsAdminService.java
package ma.zs.alc.service.facade.admin.common;

import java.util.List;
import ma.zs.alc.bean.core.common.News;
import ma.zs.alc.dao.criteria.core.common.NewsCriteria;
import ma.zs.alc.zynerator.service.IService;



public interface NewsAdminService extends  IService<News,NewsCriteria>  {

    List<News> findByCollaboratorId(Long id);
    int deleteByCollaboratorId(Long id);
    long countByCollaboratorId(Long id);




}
